saleor/graphql/core/mutations.py

```diff
@@ -157,10 +157,10 @@ def get_node_or_error(cls, info, node_id, field="id", only_type=None, qs=None):
 
         try:
             if only_type is not None:
-                pk = from_global_id_strict_type(node_id, only_type, field=field)
+                _type, pk = from_global_id_or_error(node_id, only_type, field=field)
```

Comment:
IMO, this `if` is not needed here. You can handle this by single call `from_global_id_or_error(node_id, only_type=only_typre, field=field)`